reaper/repair_run.go

```diff
@@ -0,0 +1,355 @@
+	path := fmt.Sprint("/repair_run/", repairRunId, "/state/", RepairRunStateRunning)
+	_, err := c.doPut(ctx, path, nil, nil, http.StatusOK, http.StatusNoContent)
+	if err == nil {
+		return nil
+	}
+	return fmt.Errorf("failed to start repair run %v: %w", repairRunId, err)
+}
+
+func (c *client) PauseRepairRun(ctx context.Context, repairRunId uuid.UUID) error {
+	path := fmt.Sprint("/repair_run/", repairRunId, "/state/", RepairRunStatePaused)
+	_, err := c.doPut(ctx, path, nil, nil, http.StatusOK, http.StatusNoContent)
+	if err == nil {
+		return nil
+	}
+	return fmt.Errorf("failed to pause repair run %v: %w", repairRunId, err)
+}
+
+func (c *client) ResumeRepairRun(ctx context.Context, repairRunId uuid.UUID) error {
+	path := fmt.Sprint("/repair_run/", repairRunId, "/state/", RepairRunStateRunning)
+	_, err := c.doPut(ctx, path, nil, nil, http.StatusOK, http.StatusNoContent)
+	if err == nil {
+		return nil
+	}
+	return fmt.Errorf("failed to resume repair run %v: %w", repairRunId, err)
+}
+
+func (c *client) AbortRepairRun(ctx context.Context, repairRunId uuid.UUID) error {
+	path := fmt.Sprint("/repair_run/", repairRunId, "/state/", RepairRunStateAborted)
+	_, err := c.doPut(ctx, path, nil, nil, http.StatusOK, http.StatusNoContent)
+	if err == nil {
+		return nil
+	}
+	return fmt.Errorf("failed to abort repair run %v: %w", repairRunId, err)
+}
+
+func (c *client) GetRepairRunSegments(ctx context.Context, repairRunId uuid.UUID) (map[uuid.UUID]*RepairSegment, error) {
+	path := fmt.Sprint("/repair_run/", repairRunId, "/segments")
+	res, err := c.doGet(ctx, path, nil, http.StatusOK)
+	if err == nil {
+		repairRunSegments := make([]*RepairSegment, 0)
+		err = c.readBodyAsJson(res, &repairRunSegments)
+		if err == nil {
+			repairRunSegmentsMap := make(map[uuid.UUID]*RepairSegment, len(repairRunSegments))
+			for _, segment := range repairRunSegments {
+				repairRunSegmentsMap[segment.Id] = segment
+			}
+			return repairRunSegmentsMap, nil
+		}
+	}
+	return nil, fmt.Errorf("failed to get segments of repair run %v: %w", repairRunId, err)
+}
+
+func (c *client) AbortRepairRunSegment(ctx context.Context, repairRunId uuid.UUID, segmentId uuid.UUID) error {
+	path := fmt.Sprint("/repair_run/", repairRunId, "/segments/abort/", segmentId)
+	_, err := c.doPost(ctx, path, nil, nil, http.StatusOK)
+	if err == nil {
+		return nil
+	}
+	return fmt.Errorf("failed to abort segment %v of repair run %v: %w", segmentId, repairRunId, err)
+}
+
+func (c *client) DeleteRepairRun(ctx context.Context, repairRunId uuid.UUID, owner string) error {
+	path := fmt.Sprint("/repair_run/", repairRunId)
+	queryParams := &url.Values{"owner": {owner}}
+	res, err := c.doDelete(ctx, path, queryParams, http.StatusAccepted)
+	if err == nil {
+		return nil
+	} else {
+		// FIXME this REST resource currently returns 500 for succeeded deletes
+		if res != nil && res.StatusCode == http.StatusInternalServerError {
```

Comment:
@adejanovski should I open an issue for that? I checked what the web UI does, and the error happens there as well (you see the stack trace in the logs), but the UI seems to swallow the 500 HTTP status code and gladly informs the user that the delete was successful.

Reply:
ouch, yes please 🙏

Reply:
The error is:

```
i.d.j.e.LoggingExceptionMapper - Error handling a request: 28aa7519bbc77c3d
java.lang.IllegalStateException: startTime only valid when runState is not NOT_STARTED. DELETED null
at com.google.common.base.Preconditions.checkState(Preconditions.java:737)
at io.cassandrareaper.core.RepairRun$Builder.build(RepairRun.java:303)
at io.cassandrareaper.storage.MemoryStorage.deleteRepairRun(MemoryStorage.java:258)
at io.cassandrareaper.resources.RepairRunResource.deleteRepairRun(RepairRunResource.java:761)
```

Reply:
There you go: https://github.com/thelastpickle/cassandra-reaper/issues/1090

Reply:
Thanks!